script.js

```diff
@@ -64,3 +64,41 @@ function getCounty(lat, lon) {
 // Test cases
 // getCounty(0, 0);
 // getCounty(47.487983, -121.723172);
+
+
+function popularity (trailArray) {
+    trailArray.forEach(trail => {
+        var voteCount = trail.starVotes;
+        var rating = trail.stars;
+        var popularity = "";
+        if (rating >= 4.5) {
+            voteCount += 8;
+        } else if (rating > 4) {
+            voteCount += 5;
+        } else if (rating > 3.5) {
+            voteCount += 3;
+        } else {
+            voteCount -= 3;
+        }
+
+        if (voteCount < 10) {
+            popularity = 1;
+        } else if (voteCount < 21) {
+            popularity = 2;
+        } else if (voteCount < 46) {
+            popularity = 3;
+        } else if (voteCount < 70) {
+            popularity = 4;
+        } else if (voteCount >= 70) {
+            popularity = 5
+        } else {
```

Comment:
So this else is not really doing what you want here. If you want to catch error, using try-catch statement.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch
But since here you are really getting value from an js object array and changing value of a field in each object, it should not be necessary to do error handling here.

Reply:
So I changed it and basically what we have is if for some reason the API doesn't return data for the stars or votes the popularity will default to 1 (unless it's a string like "chicken" or something but that seems unlikely) which makes sense for us. We will have to test when we are dealing with larger data sets.